torch/distributions/utils.py

```diff
@@ -23,7 +23,9 @@
 ]
 
 
-def broadcast_all(*values):
+# FIXME: Use (*values: *Ts) -> tuple[Tensor for T in Ts] if Mapping-Type is ever added.
+#   See https://github.com/python/typing/issues/1216#issuecomment-2126153831
+def broadcast_all(*values: Union[Tensor, Number]) -> tuple[Tensor, ...]:
```

Comment:
Currently, the annotation does not cover the case of  `__torch_function__`.

This is somewhat outside the scope of this PR, I think possibly a `Protocol` type should be added to `torch.types` that covers the concrete semantics.